java/src/main/native/src/ColumnVectorJni.cpp

```diff
@@ -457,6 +459,30 @@ JNIEXPORT jobject JNICALL Java_ai_rapids_cudf_ColumnVector_exactQuantile(JNIEnv
   CATCH_STD(env, NULL);
 }
 
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_normalizeNANsAndZeroes(JNIEnv *env,
+                                                                                jclass clazz,
+                                                                                jlong input_column) {
+  using cudf::column;
+  using cudf::mutable_column_view;
+  using cudf::legacy::gdf_column_to_view;
+  using cudf::legacy::view_to_gdf_column;
+
+  JNI_NULL_CHECK(env, input_column, "native handle is null", 0);
+  try {
+    std::unique_ptr<column> p_normalized_column
+        = cudf::normalize_nans_and_zeros(
+            gdf_column_to_view(*reinterpret_cast<gdf_column*>(input_column)));
+
+    auto ret = reinterpret_cast<jlong>(
+        new gdf_column(view_to_gdf_column(p_normalized_column->mutable_view())));
+
+    p_normalized_column->release();
```

Comment:
This calls release without doing anything with the return value, so this effectively deletes the contents of the column. Your `ret` will have dangling pointers to free'd memory.

Reply:
Ok, I think I misunderstood the semantics of `column::release()`. Let me try again.

Reply:
We're in something of a bind. :/ 
One tentative solution to keeping the `column::contents` alive (long enough for `gdf_column` to make it back to Java-land) might be:
```c++
    auto ret = reinterpret_cast<jlong>(
        new gdf_column(view_to_gdf_column(p_normalized_column->mutable_view())));

    auto contents = p_normalized_column->release();
    (void)contents.data.release();      // Data-buffer contents have been adopted already
    (void)contents.null_mask.release(); // by the gdf_column being returned. Relinquish ownership.

    return ret;
```
But this would leave the `device_buffer` object itself on the heap. :/

Reply:
Chaps, I was wondering if this version is more acceptable. I realize that we might leak the `device_buffer` members (the pointers, not the memory it points to). I don't see a way around it without `DeviceVector` switching over to use `column` instead of translating from `gdf_column`.

Reply:
I don't see how you're going to be able to use this libcudf++ API until the JNI machinery has been updated to use the new `cudf::column` and associated types.

Reply:
>  until the JNI machinery has been updated to use the new `cudf::column` and associated types.

I agree. The Gordian Knot tightens. :/
Would you (and @jlowe, and @revans2) be averse to this current version being checked in (to work with `gdf_column`) until we sort out our move to `cudf::column`?

Reply:
That's purely up to you guys.